test/query_test/query_test.go

```diff
@@ -50,6 +51,85 @@ var _ = Describe("Service Manager Query", func() {
 		}
 	})
 
+	Context("Service instance and last operations query test", func() {
+		var serviceInstance1, serviceInstance2 *types.ServiceInstance
+		BeforeEach(func() {
+			_, serviceInstance1 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+			_, serviceInstance2 = common.CreateInstanceInPlatform(ctx, ctx.TestPlatform.ID)
+		})
+
+		AfterEach(func() {
+			ctx.CleanupAdditionalResources()
+		})
+
+		It("should return the last operation for a newly created resource", func() {
+			queryParams := map[string]interface{}{
+				"id_list": []string{serviceInstance1.ID},
+			}
+
+			list, err := repository.QueryForList(context.Background(), types.OperationType, storage.QueryForLastOperationsPerResource, queryParams)
+			Expect(err).ShouldNot(HaveOccurred())
+			lastOperation := list.ItemAt(0).(*types.Operation)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(list.Len()).To(BeEquivalentTo(1))
+			Expect(lastOperation.State).To(Equal(types.SUCCEEDED))
+			Expect(lastOperation.Type).To(Equal(types.CREATE))
+		})
+
+		When("new operation is created for the resource", func() {
+			BeforeEach(func() {
+				operation := &types.Operation{
+					Base: types.Base{
+						ID:        "my_test_op_latest",
+						CreatedAt: time.Now(),
+						UpdatedAt: time.Now(),
+						Labels:    make(map[string][]string),
+						Ready:     true,
+					},
+					Description:       "my_test_op_latest",
+					Type:              types.CREATE,
+					State:             types.SUCCEEDED,
+					ResourceID:        serviceInstance1.ID,
+					ResourceType:      web.ServiceInstancesURL,
+					CorrelationID:     "test-correlation-id",
+					Reschedule:        false,
+					DeletionScheduled: time.Time{},
+				}
+
+				_, err := repository.Create(context.Background(), operation)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+			It("should return the new operation as last operation", func() {
+				queryParams := map[string]interface{}{
+					"id_list": []string{serviceInstance1.ID},
+				}
+				list, err := repository.QueryForList(context.Background(), types.OperationType, storage.QueryForLastOperationsPerResource, queryParams)
+				Expect(err).ShouldNot(HaveOccurred())
+				lastOperation := list.ItemAt(0).(*types.Operation)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(list.Len()).To(BeEquivalentTo(1))
+				Expect(lastOperation.State).To(Equal(types.SUCCEEDED))
+				Expect(list.Len()).To(BeEquivalentTo(1))
+				Expect(lastOperation.ID).To(Equal("my_test_op_latest"))
+			})
+
+		})
+
+		It("should return the last operation for every instances in the query", func() {
```

Comment:
I would add another operation for a different resource, and make sure that i got only the operations for the resources that were requested, and not the other one. I think at the moment this is not tested.